framework/src/application/node/forger/forger.ts

```diff
@@ -343,12 +345,12 @@ export class Forger {
 	}
 
 	public async forge(): Promise<void> {
+		const MS_IN_A_SEC = 1000;
 		const currentSlot = this._chainModule.slots.getSlotNumber();
-		const currentSlotTime = this._chainModule.slots.getRealTime(
-			this._chainModule.slots.getSlotTime(currentSlot),
-		);
 
-		const currentTime = new Date().getTime();
+		const currentSlotTime = this._chainModule.slots.getSlotTime(currentSlot);
+
+		const currentTime = Math.floor(new Date().getTime() / MS_IN_A_SEC);
```

Comment:
on the previous implementation, currentTime was in `ms`, and in below it's comparing with `currentTime <= currentSlotTime + waitThreshold`.

I think unit should be
currentSlotTime => ms
currentTIme => ms
waitThreshold =>ms

Reply:
This will be solved by using only seconds in Slots.ts so resolving.